src/ngraph/runtime/cpu/mkldnn_utils.hpp

```diff
@@ -24,6 +24,76 @@
 #include "ngraph/runtime/cpu/op/batch_norm_relu.hpp"
 #include "ngraph/type/element_type.hpp"
 
+#if MKLDNN_VERSION_MAJOR < 1
+#define FORMAT format
+#define FORMAT_KIND format
+#define FORMAT_KIND_UNDEF mkdnn_format_undef
+#define FORMAT_ANY mkldnn_any
+#define FORMAT_UNDEF mkldnn_undef
+#define DATA_UNDEF data_undef
+
+#define CHANGE_FORMAT                                                                              \
+    if (weights_desc.data.format == mkldnn_nchw)                                                   \
+    {                                                                                              \
+        weights_desc.data.format = mkldnn_oihw;                                                    \
+    }                                                                                              \
+    if (weights_desc.data.format == mkldnn_ncdhw)                                                  \
+    {                                                                                              \
+        weights_desc.data.format = mkldnn_oidhw;                                                   \
+    }
+
+#define BN_FLAG_CLASS batch_normalization_flag
+
+#define PADDING                                                                                    \
+    , mkldnn::padding_kind::zero)
+
+#define SET_ROUND_MODE attr.set_int_output_round_mode(mkldnn::round_mode::round_nearest);
+
+#define QUERY_SCRATCHPAD_SUM(X)                                                                    \
+    {                                                                                              \
+    }
+
+#define QUERY_SCRATCHPAD_CONCAT(X)                                                                 \
+    {                                                                                              \
+    }
+
+#else
+#define TENSOR_MAX_DIMS MKLDNN_MAX_NDIMS
+#define FORMAT format_tag
+#define FORMAT_KIND format_kind
+#define FORMAT_KIND_UNDEF format_kind::undef
+#define FORMAT_ANY static_cast<mkldnn_format_kind_t>(mkldnn::memory::format_kind::any)
+#define FORMAT_UNDEF format_tag::undef
+#define DATA_UNDEF undef
+
+#define CHANGE_FORMAT                                                                              \
+    {                                                                                              \
+    }
+
+#define BN_FLAG_CLASS normalization_flags
+
+#define PADDING                                                                                    \
+        )
+
+#define SET_ROUND_MODE                                                                             \
+    {                                                                                              \
+    }
+
+#define QUERY_SCRATCHPAD_SUM(X) mkldnn_emitter->query_scratchpad_sum(X);
```

Comment:
One macro that takes the func name + variadic args list would be better : `QUERY_SCRATCHPAD(sum, x)`
nit: wrap in do/while(0)